File: Behavioural-Patterns/Mediator/Java/Implementations/LoggingService.java
package Implementations;

import Interfaces.Subscriber;

public class LoggingService implements Subscriber {

    @Override
    public void onEvent(Object event) {
        if (event instanceof UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) event;
            System.out.println("   [LoggingService] Logged event: User '" + loginEvent.username + "' logged in at " + loginEvent.timeStamp);
        }
    }

    @Override
    public String getName() {
        return "LoggingService";
    }
    
}

File: Behavioural-Patterns/Mediator/C#/Mediator/Implementations/LoggingService.cs
public class LoggingService : Subscriber {

    public void onEvent(Object ev) {
        if (ev is UserLoggedInEvent) {
            UserLoggedInEvent loginEvent = (UserLoggedInEvent) ev;
           Console.WriteLine("   [LoggingService] Logged event: User '" + loginEvent.username + "' logged in at " + loginEvent.timeStamp);
        }
    }

    public String getName() {
        return "LoggingService";
    }
    
}
